libconcurrent/primitives/threadtools.c

```diff
@@ -19,24 +19,40 @@ inline static void *kthreadWrapper(void *arg);
 
 static __thread pthread_t *__threads;
 static __thread int32_t __thread_id = 0;
-static __thread int32_t __prefered_core = 0;
+static __thread int32_t __preferred_numa_node = 0;
+static __thread int32_t __preferred_core = 0;
 static __thread int32_t __unjoined_threads = 0;
 
 static void *(*__func)(void *) CACHE_ALIGN = NULL;
 static uint32_t __uthreads = 0;
 static uint32_t __nthreads = 0;
 static uint32_t __ncores = 0;
+static uint32_t __schedule_policy = SYNCH_THREAD_PLACEMENT_DEFAULT;
 static bool __uthread_sched = false;
 static bool __system_oversubscription = false;
 static bool __noop_resched = false;
 static SynchBarrier bar CACHE_ALIGN;
 
+
+void synchSetThreadPlacementPolicy(uint32_t policy) {
+    __schedule_policy = policy;
```

Comment:
A check that the policy value is within limits maybe useful here
'policy <= max_policy_value'

Reply:
OK, I will add a check here.

Reply:
Done!